spec/requests/posts_spec.rb

```diff
@@ -2,9 +2,34 @@
 
 RSpec.describe 'Posts', type: :request do
   describe 'GET /index' do
-    it 'returns http success' do
-      get '/posts/index'
-      expect(response).to have_http_status(:success)
+    before(:each) do
+      get '/users/1/posts'
+    end
+
+    it 'Users index is successful' do
+      expect(response).to have_http_status(:ok)
+    end
+    it 'Renders template correctly' do
+      expect(response).to render_template(:index)
+    end
+
+    it 'Response should include right placeholder text' do
+      expect(response.body).to include('List of all post for a given user')
+    end
+  end
+
+  describe 'Show single user details' do
+    before(:each) do
+      get '/users/1/posts/1'
+    end
+    it 'Response will be successful.' do
+      expect(response).to have_http_status(:ok)
+    end
+    it 'Template render correctly' do
+      expect(response).to render_template(:show)
+    end
+    it 'Response should include right placeholder text.' do
+      expect(response.body).to include('Post details will be shown here')
```

Comment:
- [ ] Kindly make sure that tests written for index and show actions for User and Post controllers are passing, the tests are currently failing see screenshot below ⬇️  


![Screenshot_511](https://user-images.githubusercontent.com/69638013/198247687-fbf8b06c-f3bf-4c8b-bfad-d21ec7eb295a.png)